Libs/Mesh/Mesh.cpp

```diff
@@ -105,148 +99,696 @@ bool Mesh::write(const std::string &pathname)
       writer->SetFileName(pathname.c_str());
       writer->SetInputData(this->mesh);
       writer->Update();
+      return *this;
     }
 
     if (StringUtils::hasSuffix(pathname, ".vtp")) {
       auto writer = vtkSmartPointer<vtkXMLPolyDataWriter>::New();
       writer->SetFileName(pathname.c_str());
       writer->SetInputData(this->mesh);
       writer->Update();
+      return *this;
     }
 
     if (StringUtils::hasSuffix(pathname, ".stl")) {
       auto writer = vtkSmartPointer<vtkSTLWriter>::New();
       writer->SetFileName(pathname.c_str());
       writer->SetInputData(this->mesh);
       writer->Update();
+      return *this;
     }
 
     if (StringUtils::hasSuffix(pathname, ".obj")) {
       auto writer = vtkSmartPointer<vtkOBJWriter>::New();
       writer->SetFileName(pathname.c_str());
       writer->SetInputData(this->mesh);
       writer->Update();
+      return *this;
     }
 
     if (StringUtils::hasSuffix(pathname, ".ply")) {
       auto writer = vtkSmartPointer<vtkPLYWriter>::New();
```

Comment:
Should this be using @akenmorris 's thread safe PLY reader/writer? Not a problem for the command line tools, but for the Python API

Reply:
Probably. How difficult will it be to make that change?

Reply:
@akenmorris thoughts on this?

Reply:
Well, I contemplated putting the lock there for everything, but I hesitated to put a lock in the Mesh constructor because I do not know how it will behave in all cases (e.g. called from Python in multiple threads?)